src/os.rs

```diff
@@ -143,11 +143,16 @@ impl ReadRng {
         let mut guard = mutex.lock().unwrap();
         if (*guard).is_none() {
             info!("OsRng: opening random device {}", path.as_ref().display());
-            let file = File::open(path).map_err(|err| Error::with_cause(
-                    ErrorKind::Failure,
-                    "error opening random device",
-                    err
-            ))?;
+            let file = File::open(path).map_err(|err| {
+                use std::io::ErrorKind::*;
+                match err.kind() {
+                    NotFound | PermissionDenied | ConnectionRefused =>
```

Comment:
This does not really match the same posix errors, see https://github.com/rust-lang/rust/blob/master/src/libstd/sys/unix/mod.rs#L96. I don't want to be picky, but think it is better to match those (can be extracted).

Reply:
I don't get you? The names are all basically the same except for `ENOENT` and `EAGAIN`. You think we should match the constants defined in `libc` instead? Or you would choose a different selection of constants to map to `Unavailable`?

Reply:
I don't think `ConnectionRefused` is an error we can encounter for `OsRng`, and am missing `EMFILE`, `ENFILE`, `ENODEV` and `ENXIO`. Interesting, if you search for ["ENXIO random"](https://www.google.com/search?q=ENXIO+random) python uses a similar list when opening `/dev/random`. Maybe using the `libc` constants is best here?